.circleci/config.yml

```diff
@@ -1,8 +1,408 @@
+          key: 'go-cache-{{ .Environment.CIRCLE_STAGE }}-{{ .Environment.CIRCLE_BUILD_NUM }}'
+          paths:
+            - tmp/go-cache
+      - persist_to_workspace:
+          root: << parameters.circleci_home >>
+          paths:
+            - project
+            - go
+            - gimme
+            - .gimme
+
+  cache_libsodium:
+    description: Cache libsodium for build
+    steps:
+      - run:
+          name: Get libsodium md5
+          command: |
+            mkdir -p tmp
+            find crypto/libsodium-fork -type f -exec openssl md5 "{}" + > tmp/libsodium.md5
+      - save_cache:
+          key: 'libsodium-fork-{{ .Environment.CIRCLE_STAGE }}-{{ checksum "tmp/libsodium.md5" }}'
+          paths:
+            - crypto/libs
+
+  restore_libsodium:
+    description: Restore libsodium for build
+    steps:
+      - run:
+          name: Get libsodium md5
+          command: |
+            mkdir -p tmp
+            find crypto/libsodium-fork -type f -exec openssl md5 "{}" + > tmp/libsodium.md5
+      - restore_cache:
+          keys:
+            - 'libsodium-fork-{{ .Environment.CIRCLE_STAGE }}-{{ checksum "tmp/libsodium.md5" }}'
+
+  nightly_test:
+    description: Run nightly tests from build workspace, for re-use by diferent architectures
+    parameters:
+      circleci_home:
+        type: string
+        default: "/home/circleci"
+      result_subdir:
+        type: string
+    steps:
+      - attach_workspace:
+          at: << parameters.circleci_home >>
+      - run: mkdir -p /tmp/results/<< parameters.result_subdir >>
+      - restore_cache:
+          keys:
+            - 'go-cache-{{ .Environment.CIRCLE_STAGE }}-'
+      - run:
+          name: Run nightly tests
+          no_output_timeout: 45m
+          command: |
+            set -e
+            set -x
+            export PATH=$(echo "$PATH" | sed -e 's|:<< parameters.circleci_home >>/\.go_workspace/bin||g' | sed -e 's|:/usr/local/go/bin||g')
+            export KMD_NOUSB=True
+            export GOPATH="<< parameters.circleci_home >>/go"
+            export PATH="${PATH}:${GOPATH}/bin"
+            export ALGORAND_DEADLOCK=enable
+            GOLANG_VERSION=$(./scripts/get_golang_version.sh)
+            eval "$(~/gimme "${GOLANG_VERSION}")"
+            scripts/configure_dev.sh
+            scripts/buildtools/install_buildtools.sh -o "gotest.tools/gotestsum"
+            PACKAGES="$(go list ./... | grep -v /go-algorand/test/)"
+            export PACKAGE_NAMES=$(echo $PACKAGES | tr -d '\n')
+            export PARTITION_TOTAL=$CIRCLE_NODE_TOTAL
+            export PARTITION_ID=$CIRCLE_NODE_INDEX
+            export PARALLEL_FLAG="-p 1"
+            gotestsum --format testname --junitfile /tmp/results/<< parameters.result_subdir >>/results.xml --jsonfile /tmp/results/<< parameters.result_subdir >>/testresults.json -- --tags "sqlite_unlock_notify sqlite_omit_load_extension" -race -timeout 1h -coverprofile=coverage.txt -covermode=atomic -p 1 $PACKAGE_NAMES
```

Comment:
You should find a way to add the junitfile option in the Makefile. Does this somehow trigger rebuilds when the circleci task is split across multiple workers?

In particular, the `gotestsum` command/arguments are exported in the `GOTESTCOMMAND` variable at the top of the Makefile, and that variable is used later on when running integration tests.

If I'm correct, this means that the integration tests fall back to using `go test`, and there is no report generated for the e2e tests.